### ISSUE TITLE
Cross-repository milestones/versions

### ISSUE THREAD

**Issue by iBeliever** [dear-github/dear-github#93]

**Comment by elifoster** [comment]
This is somewhat achievable through [Waffle](https://waffle.io/) by adding all of your repositories to a single "Waffle". However I don't think it supports milestones very well; it just displays them on the issue cards.

**Comment by softmonkeyjapan** [comment]
Following what you've said, I would some much love to be able to have milestones depending on other milestones (e,g, `release-1.2.0` that could have it's own milestones, a.k.a `sprint`, sort of).

**Comment by iBeliever** [comment]
@elifoster I use Zenhub, which I think is similar. You can have shared milestones across projects, but you can't have sub-milestones, which is a big part of this. 

My ideal milestone layout would be:

    OS Release 0.2
     |- OS-level issue
     |- OS-level issue
     |- Project milestone
     |   |- Project-level issue
     |- Project milestone
         |- Project-level issue

**Comment by jamen** [comment]
Just ran into this on a group project I'm working in.  I wanted a cross-repository milestone for our beta launch, since you can also attach due dates on them.  I think this would be a very useful feature.

**Comment by stephenwithav** [comment]
This would be very much appreciated. +1

**Comment by garcianavalon** [comment]
This would be great, its a pain to orchestrate a release between several components +1

**Comment by koenpunt** [comment]
Projects are already available organization-wide; https://github.com/orgs/NAME/projects

**Comment by mcqueary** [comment]
The new Projects feature is great. If only Milestones and Projects could be tied together. Or Milestones could include issues from multiple repos. Or something :)

**Comment by kareem** [comment]
> What we really need is organization-wide milestones where we can add issues or milestones from any project in the organization.

Codetree supports this out of the box. You create what we call a "project" in Codetree that rolls up issues from multiple GH repos.  Then, when you create a milestone in the project, milestones are automatically created in every GH repo that belongs to that project.  

The net effect is that you can add issues from different repos to the same milestone with basically zero pain.

**Comment by base698** [comment]
Would be super nice.

**Comment by obengwilliam** [comment]
This will be very helpful

**Comment by ldhasson** [comment]
Would love this too... Adding an external tool such as CodeTree is not what we'd be looking for. We have the same issue where different teams work on different independent repositories, but often, an issue in a component needs a fix in another component. So, allowing issues to span multiple repos would be so helpful. At the very least, documenting and enforcing some dependency, i.e., Issue #248 in Project X depends on issue #445 in Project Y and #1109 in Project Z would allow form better tracking of these things.

Additionally, extending this behavior to milestones and releases would be nice to have.

**Comment by imincik** [comment]
This is very much needed feature for my organization as well.

**Comment by QAnders** [comment]
We love Projects, super handy!
If we could only get the milestones across repositories too it would be great!

**Comment by mccrodp** [comment]
Instead of creating Milestones you can create a "Project" as a milestone. You can also add Projects to Projects when nesting / sub projects is needed. So, in the case where you need a cross repo Milestone, just create a Project instead. Works for my use case anyway 🚀 😄

**Comment by clarkbw** [comment]
I recommend using projects. It’s more likely that the features that make milestones interesting will come to projects than the other way around.

**Comment by gamontoya** [comment]
Dear Github: can you tell us if this feature is being considered? Sincerely, me.

**Comment by mcritchlow** [comment]
Our org would really benefit from this too. Having org-wide projects implies you have a team, or multiple teams, working on a cross section of repos in a given iteration/sprint. This works great now that we have org-wide projects, thank you!. But that context pretty naturally lends itself to wanting to have Milestones that also cut across these same repos. A "Sprint 12 - 2018" milestone, for example, that ties all the tickets from a given iteration together, would be most helpful. As it is now, we have to create the same Milestone in each repository so we can do an aggregate query with the API. This works.. but the UX would be much better if this can be directly integrated into org-wide projects. Thanks for considering!

**Comment by nilsnolde** [comment]
Yes, pls!!

**Comment by varunest** [comment]
This is exactly I was looking for my organization. It would be very helpful if we could create project-wise milestone which could reference issues from different repositories.

**Comment by aniesshsethh** [comment]
Yep, this will be very helpful

**Comment by DylanThomson** [comment]
Development team of 30+ Devs here also wanting this! A milestone having scope only of a single code repository is such a dated idea. What kind of project is run from a single monolithic code base in 2018?
A single project for us is spread across many agnostic modules (Upward of 10 repos can be used to build a single app (excluding backend)) This makes the current implementation of milestones basically useless.

**Comment by svnpenn** [comment]
https://github.com/dear-github/dear-github/issues/209

**Comment by svnpenn** [comment]
You can do this with projects:

1. Click "Add a note to this column"
2. Paste link to issue from another repo
3. Click "Add"

http://help.github.com/articles/about-project-boards

Alternatively if you create an organization you can add cards for issues from
any repo owned by the org.

- http://help.github.com/articles/creating-a-new-organization-from-scratch
- http://help.github.com/articles/managing-project-boards-in-your-repository-or-organization

**Comment by AlexanderIvanof** [comment]
As temporary solution we've created simple Groovy script to create milestones via GitHub API:
1. Http GET `https://github.com/api/v3/orgs/<MyOrg>/repos` - will return all repositories
2. Http POST `https://github.com/api/v3/repos/<MyOrg>/<MyRepo>/milestones`
with body:
```json
{
  "title": "v1.0.3",
  "state": "open",
  "description": "Bug fixes for previous release",
  "due_on": "2018-10-09T00:00:00Z"
}
```
[Source here](https://developer.github.com/v3/issues/milestones/)

**Comment by PierrePX** [comment]
Yes, very much needed !

**Comment by drew-arux** [comment]
I came here because I couldn't believe organization-wide milestones weren't implemented. This severely limits their usefulness for my team. Bumped!

**Comment by controlade** [comment]
Would appreciate your implementation of Cross-Repository Milestones, thanks.

**Comment by maxvisser** [comment]
Could I add  to this issue that I want to see the milestones on the projectboards cards aswell. I currently fixed this by creating a 'organisation' repo with submodules to specific parts of my application. This way I can have the milestones in one place but the code in multiple places as I collect all issues in one repo and use labels.

**Comment by kaklik** [comment]
This feature is needed for our team also.

**Comment by edanapear** [comment]
Our team would also benefit from being able to add project-level milestones – accessible by reference across the linked repos.

**Comment by fsadykov** [comment]
This is a very nice feature please implement that. I am trying to use milestone for all repositories but I am not able to that right now. :((

**Comment by jsduffy** [comment]
This would also be useful for our team!

**Comment by mpvader** [comment]
pls make! we need this too. Projects are nice, but overkill for our need.

**Comment by BenRCopilot** [comment]
Like the armies of other people said, this would be hugely valuable. We're replacing JIRA entirely with Github and this would reduce one other pain point

**Comment by controlade** [comment]
Hi @BenRCopilot 
What was your main reason for moving from JIRA to GitHub.

I use GH in my organization, and was evaluating the upsides of moving to JIRA.
Thanks in advance for any comment you would like to share on this subject.

**Comment by controlade** [comment]
> @controlade this would be way easier to explain over a phone or video call, but essentially it came down to engineers are not going to manage JIRA workflows very much and it became a lot of overhead for product to maintain and frankly wasn't all that helpful in reporting (Burndown) etc. bc too many things went unlogged in JIRA. It was a partial view and JIRA doesn't do a good job speaking with other SaaS systems. We have created a 2 way synch between Productboard (where product writes stories and acceptance criteria) <-> Github (where Issues+labels map to tasks, bugs, stories etc) <-> Zendesk or Freshdesk (Ticketing+reporting) to feed into our Product backlog. I was started to look at Slackbots to automate JIRA flows, but frankly Github had automation built into preconfig Kanban boards.

Thanks you for your reply.  To share our point of view, we prefer simple, lean tooling against lots of features. But, Cross-repository milestones would help a lot.

**Comment by bauwaerter** [comment]
+1

**Comment by DanielSeehausen** [comment]
+1

**Comment by blackfalcon** [comment]
:+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: 

OMG ... YES! I have multiple repos I work with across teams and we all share the same release dates. 

Why is there not an option to manage milestones from the PROJECT level? That only makes sense.

**Comment by willemgons** [comment]
We have just walked ourselves into the same situation.
Milestones for multiple repos is a must.
Individual issues with an Estimated Time would ALSO be useful IF it would roll up to be reflected in the milestone.

**Comment by NTonani** [comment]
+1 Milestones need a makeover to stay relevant with current best practices and standards

**Comment by tobi-v** [comment]
That would be super useful

**Comment by maxvisser** [comment]
Just to give an update: We fixed this by using Zenhub for creating milestones. Its pretty good at keeping milestones insync. Only downside if you change a title; it will close the old milestone and create a new one; so when creating the milestone you should probably stick with it ;)

**Comment by mayrop** [comment]
Please please please: Milestones for multiple repos would make our life much easier!!

**Comment by jeanmichaelgeorg** [comment]
Same here, life would be better with cross repo milestones and labels

**Comment by isglass** [comment]
+1

**Comment by holubky** [comment]
+1

**Comment by Salles-FA** [comment]
+1

**Comment by tkrempser** [comment]
+1

**Comment by mtsmachado8** [comment]
+1

**Comment by lusu007** [comment]
+1

**Comment by AparnaDhinakaran** [comment]
+1

**Comment by kiranmr** [comment]
Same here, milestones that work across repo will be very useful.

**Comment by n00bsys0p** [comment]
Agree - ideally we would be able to assign milestones & labels at the Project level, which would simplify Github Project management significantly.

**Comment by blackfalcon** [comment]
This thread was started in 2016. Github ... do you have any intention to solve this issue? It's a wildly popular request and there has been no response from you.

**Comment by eedduuar** [comment]
+1

**Comment by egluhbegovic** [comment]
+1

**Comment by susheel-clearcals** [comment]
+1

**Comment by styrbjornp** [comment]
+1

**Comment by adamdry** [comment]
As an only-partial workaround, we create the same label across all repos (same name, same colour) and apply them to Issues for that "milestone". This allows us to visually scan a Project board and see Issues that relate to the same "milestone".

It doesn't give you progress reporting unfortunately - other than scanning the Project board with your eyes.

**Comment by blackfalcon** [comment]
@adamdry we have done the same thing. https://github.com/Financial-Times/github-label-sync has been a HUGE help with keeping labels in sync across repositories. With that, we also made a series of dashboards using the Github GraphQL API. https://auro.alaskaair.com/planned-work

**Comment by brockers** [comment]
How is this still a thing?  Seriously, how many large projects only have a single repo as part of their product milestones?

+1

**Comment by briandoyle81** [comment]
This would be a very useful feature.  +1

**Comment by mpdougherty** [comment]
+1

**Comment by peterinnesmsft** [comment]
+1

**Comment by jfallows** [comment]
+1

**Comment by sl1g18** [comment]
+1 This and organization wide labels would make GitHub Projects much more useful!

**Comment by blackfalcon** [comment]
@sl1g18 check this out https://www.npmjs.com/package/github-label-sync

It's not the best solution, but it's the solution we have. It works pretty well, but native support would be better. Agreed.

**Comment by mat-work** [comment]
+1

**Comment by dpetrovaliev** [comment]
Would be great to have shared milestones between multiple repositories in the organisation.

**Comment by andresovela** [comment]
+1

**Comment by georgii-ivanov** [comment]
+1

**Comment by victormartinez-work** [comment]
+1

**Comment by tomassykora** [comment]
+1

**Comment by WarGreyGon** [comment]
+1. 
In a real life project, you will be able to complete a milestone by implementing features on multiple applications.

How can you track your milestone progress if you cannot include all the related issues from all related repositories?

Even having the same milestone (same name) accross multiple repos is a mess

**Comment by hassanctech** [comment]
+1 Would love to see this!

**Comment by fecetrulo** [comment]
+1 Would love this!!

**Comment by etienne-napoleone** [comment]
Project management using github will be lackluster until this is implemented ...

**Comment by wogsland** [comment]
+1

**Comment by pbochynski** [comment]
Until the feature is implemented in the UI you can use GitHub API to sync milestones between multiple repositories (or even organizations). With GitHub java-script client you can do that in a few lines of code. I created a simple script for managing milestones and labels across all repositories in the kyma-project organization. The script is published here: https://github.com/pbochynski/github-sync

**Comment by MariusMeiners** [comment]
+1

**Comment by Salles-FA** [comment]
Hi,

GitHub devs are announcing this new feature "Discussions" and I posted a new discussion at the official repo.

Please, let's move this discussion over there!

Thx all!

**Comment by Salles-FA** [comment]
https://github.com/github/feedback/discussions/2945

**Comment by lschnoller** [comment]
much appreciated +1

**Comment by tela** [comment]
Looks like Discussions is a feature, and this issue is probably still the best place for us track this request.

**Comment by HenKun** [comment]
> Instead of creating Milestones you can create a "Project" as a milestone. You can also add Projects to Projects when nesting / sub projects is needed. So, in the case where you need a cross repo Milestone, just create a Project instead. Works for my use case anyway 🚀 😄

@mccrodp Can you elaborate on how to add a project to a project, please?

**Comment by rohanrajpal** [comment]
+1 would love to have this feature

**Comment by MuriloSchaefer** [comment]
would love this feature

**Comment by bakayolo** [comment]
+1, I started playing with github projects and immediately search for org milestone!

**Comment by vlad-troyan** [comment]
+1

**Comment by cspavlou** [comment]
+1 - This would be GREAT!

**Comment by moonso** [comment]
+1

**Comment by wtfabio** [comment]
+1!

**Comment by jvedev** [comment]
+1

**Comment by bqchristie** [comment]
+1

**Comment by jpvajda** [comment]
+ 💯

**Comment by BrandonEdwards** [comment]
I'll add a +1 to the masses begging for this feature.

**Comment by gezquinndesign** [comment]
5 years on, here's one more user who'd love this. 👍

**Comment by leonardoviada** [comment]
It would be a great advantage in multi-repo planning

**Comment by gezquinndesign** [comment]
Do the github engineers see this?

**Comment by jpvajda** [comment]
@gezquinndesign brings up a good a question, so I sent this issue to github in the form of a support ticket. 
<img width="1023" alt="Screen Shot 2021-03-16 at 9 17 34 AM" src="https://user-images.githubusercontent.com/1395158/111333708-8ace5280-8638-11eb-8c06-5750a60cbd7a.png">

**Comment by Salles-FA** [comment]
Hi @jpvajda,

Could you post the direct link to your issue, please?

**Comment by jpvajda** [comment]
@Salles-FA well it's behind authentication so you couldn't access the actual ticket, but here is the url

https://support.github.com/ticket/personal/0/1065333

**Comment by adriangroch** [comment]
+1 CC: @dl-pendula

**Comment by Arco-Dielhof** [comment]
+1

**Comment by fevb** [comment]
+1

**Comment by Tibalt** [comment]
+1

**Comment by gezquinndesign** [comment]
> @Salles-FA well it's behind authentication so you couldn't access the actual ticket, but here is the url
> 
> https://support.github.com/ticket/personal/0/1065333

Any response from support? @Salles-FA

**Comment by Salles-FA** [comment]
Hi @gezquinndesign ,

> Any response from support? @Salles-FA

@jpvajda has created that...

**Comment by gezquinndesign** [comment]
Hi @Salles-FA, sorry got mixed up there. 🙂 I guess your @ should prompt a response.

**Comment by HideakiSago** [comment]
I want to do Scrum on GitHub only.
Organization's Project is a great tool!
It is now possible to filter issues across multiple repositories by giving each repository milestone the same name.
However, it is not possible to bulk change issue milestones across the repository.
I'm assuming a use case where issues are bulk moved from Backlog milestone to Sprint n.
I believe that cross-repository milestones are needed to achieve this.

**Comment by dlucian** [comment]
sooo... it's been 5 years since this was widely requested... and still there are no organization-level milestones 😥

**Comment by gezquinndesign** [comment]
Following advice given here: https://github.community/t/where-do-i-post-feature-requests-for-github/1501, I've submitted a request via https://support.github.com/contact/feedback with a link back to this post/thread.

**Comment by hardiksondagar** [comment]
We recently moved from Asana to Github Projects, but we are using multiple repos so creating sprints across repos was painful. I did some hacking using Github API to create milestones/sprints in multiple repos at once.

Checkout here - https://hardiksondagar.github.io/cross-repository-milestones/
It requires personal access token of Github which is very easy to get.

Note - I'm not storing/stealing any kind of token on my end. It's pure frontend api calls. Do checkout code and feel free to modify it.

**Comment by sensingturtle** [comment]
+1 for this feature. Since org project boards let us use cards from all repos, I think it’s only natural for teams to implement milestone workflows that track issues across multiple repos.

**Comment by ryanmarshallmc** [comment]
pls

**Comment by nkpremices** [comment]
+1 for this feature.

**Comment by mojitoj** [comment]
+1

**Comment by alecov** [comment]
This feature is an absolute must-have, and it kinda surprises me it doesn't exist yet.

**Comment by larsfjerm** [comment]
Tracked in: https://github.com/github/roadmap/issues/276

**Comment by stefaniamak** [comment]
Now with the new org projects that GitHub added, this feature would be extremely useful! I want to sort my cross projects issues into Stories, but I can't find a useful similar sorting method.

**Comment by BirgitPohl** [comment]
I'm coming here from Google, because I have several organisations needing this feature.
Especially with cross-functional or domain teams working on some micro-service structures or tooling that depend on each other, this feature would come in very handy.

**Comment by LeonardAukea** [comment]
Glad to see this feature on the roadmap at least. Do you know about when this feature will be implemented. It's kind of the missing piece for us to go all in GitHub.

**Comment by imqdee** [comment]
Hello from Ledger 👋 
This feature would be extremely useful for us, it would simplify some of our workflows. Excited to see it in the roadmap!

**Comment by sulemanhasib43** [comment]
Looking forward to this feature 👀

**Comment by Ackak12** [comment]
We are using milestones almost equivalent to features in an agile project. And we have module codes stored in different repositories. Sometimes there are issues created in different repositories which is associated to one feature. But we can't map them to same milestone as cross repository milestones are not supported by GitHub now.
So it would be a great solution if added in future release.

**Comment by xdaluca** [comment]
looking forward to this!

**Comment by MorganeLecurieux** [comment]
Same here ! It's the main feature that is missing for us at Lago to be able to really use the Project management tool.
Right now we have to use an external tool because there is no way to link issues from different repo to a same Feature aka Milestone

**Comment by philknows** [comment]
Adding to the bumps since 2016.... because we need it.

**Comment by RengokuDK** [comment]
Super idea. Need it

**Comment by ghost** [comment]
Up-vote also on https://github.com/orgs/community/discussions/6296 in the [official  GitHub feedback discussions](https://github.com/community/community). See also [update on current status](https://github.com/orgs/community/discussions/6296#discussioncomment-3147914).

**Comment by JoshuaKGoldberg** [comment]
FYI, looks like that link is now https://github.com/community/community/discussions/6296 ✨

**Comment by nguyendon** [comment]
+1, we would really like this on my teams

**Comment by fionayz** [comment]
+1

**Comment by ppscvalentin** [comment]
+1

**Comment by djfinnoy** [comment]
+1, can't even use milestones until this is in place

**Comment by gustavogarayalde** [comment]
+1

**Comment by etienne-napoleone** [comment]
please thumbs up the main post instead of posting +1
you (and I :D) are sending a notification to dozens of people every time

**Comment by brunocalmels** [comment]
Big +1 to this issue.

**Comment by leon0399** [comment]
With introduction of Projects, this feature is just mandatory 😅

**Comment by Gictorbit** [comment]
it would be great to have milestones for whole repos in org it's like user story for all tasks

**Comment by brothercorvo** [comment]
Ples github, we need it

**Comment by ghost** [comment]
@iBelieve May you add a hint like "Vote in the official GitHub-community  at https://github.com/orgs/community/discussions/6296, too" to [initial issue-comment](https://github.com/dear-github/dear-github/issues/93#issue-127588433) you've created, please?

That all readers of this issue quickly see and upvote there. I think the GitHub employees watch that community-discussions, but (perhaps) not this one. But search-engines rank this issue highest instead.

Thanks 😄

**Comment by ollejernstrom** [comment]
Is there any update on this in 2023?

The issue is seven years old now...

Would really need it!

**Comment by savtux** [comment]
We're trying out Github Projects and the inability to have Milestones across multiple repositories is a **huge** miss. Possibly a dealbreaker.

**Comment by gtupak** [comment]
@savtux same with us!

**Comment by karmicdude** [comment]
+1

**Comment by dbrasseur-aneo** [comment]
+1

**Comment by NesManrique** [comment]
+1

**Comment by maurom88** [comment]
Please let us know if there is any intention to integrate cross-repository milestones, this would be highly beneficial for our org.

**Comment by edinella** [comment]
+1

**Comment by willemgons** [comment]
+1. Long outstanding.

**Comment by trackrx** [comment]
+1000000

**Comment by f-rakete** [comment]
![image](https://github.com/dear-github/dear-github/assets/69428398/df8d2ab4-cfb5-4f04-b7c8-e172a2886e71)

cmon

**Comment by BillChirico** [comment]
+1

**Comment by cktricky** [comment]
+1

**Comment by largek** [comment]
Aaaaand into 2024...still waiting. Require within- and across-organisation milestone tracking, i.e. issues from various repos in multiple organisations can be assigned to a milestone in another repo. Please, this would be a gamechanger.

**Comment by HenKun** [comment]
It won't happen, GitHub should honestly close the issue

**Comment by ptrkstr** [comment]
I would find this feature very useful 🙏

**Comment by awentzel** [comment]
I would even go one step further and suggest Milestones and Labels could both be created at the Enterprise level and inherited by every Organization associated with that enterprise which would work very well for service companies/agencies.

**Comment by DylanThomson** [comment]
Yes,  I've been waiting on this for about 7 years now...

**Comment by gtupak** [comment]
Maybe we should also upvote the suggestion in the official github community forum: https://github.com/orgs/community/discussions/5266

**Comment by richardhensman123** [comment]
+1

**Comment by DrHouston** [comment]
Yep- I am currently manually entering the same milestones across repositories- big headache

**Comment by MinuraPunchihewa** [comment]
+1

**Comment by f-rakete** [comment]
+1

**Comment by sfreudenthaler** [comment]
+1 🙏🙏🙏

**Comment by YuriyVelichkoPI** [comment]
+1, Creating milestones and maybe even labels on the organization-level project will be super useful. 

For now, I have to duplicate them in each repo managed by the project. 

Thanks!

**Comment by timreibe** [comment]
+1

**Comment by cktricky** [comment]
+1

**Comment by Bide-UK** [comment]
Again Please make this a thing

**Comment by Sivayogeith** [comment]
we need this!

**Comment by LucaMarconato** [comment]
+1 on this, it would be super useful!

**Comment by djcesar** [comment]
+1

**Comment by ghost** [comment]
+!

**Comment by Andre-Medina** [comment]
+1, such a bad limitation, I dont understand why you can have organization wide projects, but not organization wide milestones

**Comment by f-rakete** [comment]
time for me to do my monthly +1

but realistically they dont give 2 shits about this, the issue is quickly aproaching it's 10th year anniversary

**Comment by awentzel** [comment]
I like that idea. Every month plus 1. This is huge for any Enterprise that has multiple organizations and each has multiple projects. Sometimes projects needs migrated, sometimes accidents happen where issues are created in wrong org and you just want a simple way to transfer. It's not a a major dev challenge.

**Comment by shaur-k** [comment]
+1 for milestones across the full organization

**Comment by shessenauer** [comment]
+1!

**Comment by andrasbarabas** [comment]
+1

**Comment by arjan0307** [comment]
+1

**Comment by fernandoseguim** [comment]
+1 for milestones across the full organization

**Comment by Bide-UK** [comment]
6 Months till the 10th birthday of this request. 🎂

**Comment by cdierkens** [comment]
> 6 Months till the 10th birthday of this request. 🎂

Celebrate

**Comment by neilime** [comment]
Insane... such a massive requested feature, dozens of clients, still not implemented. Sadly this is the case for several feature and issues across the existing GitHub ecosystem.

Meanwhile we are hyped with dozen of new AI models, announcements and Copilot footages every day!

**Comment by kemptonburton** [comment]
+1

**Comment by willemgons** [comment]
We are still hopping back and forth between 3 massive repos. Prevention of cross-contamination vs ease of work.

**Comment by isasmendiagus** [comment]
+1

**Comment by danrosen25** [comment]
+1
